### PR TITLE
Perfect

### PR DESCRIPTION
**Pull request by bozoweed**
hi have finish to debug you code it's now work perfect ( for the line 45 in main.js they have my git forget to change it sorry )

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -41,8 +41,8 @@ let maxStats = {
 };
 const firewalls = ["1", "2", "3"];
 const ocrApiKey = "XXX";
-const db = "https://raw.githubusercontent.com/snollygolly/sourceio-automation/master/db.json";
-let message = "papa bless, one love /r/javascript";
+const db = "https://raw.githubusercontent.com/bozoweed/sourceio-automation/master/db.json";
```
Comment (snollygolly) [inline main.js:44]: I don't want the master version of my app to point to your fork for the DB

File: main.js [hunk 2]
```diff
@@ -41,8 +41,8 @@ let maxStats = {
...
+let message = "papa bless, one love ";
```
Comment (snollygolly) [inline main.js:45]: Also, why did you change my message?

File: main.js [hunk 3]
```diff
@@ -121,8 +121,7 @@ app = {
 				// check to see if it's new
 				if (hackProgress === newHackProgress) {
 					// the bar hasn't moved
-					log("* Progress bar hasn't moved, waiting");
-					// TODO: make this an automatic process
+					app.restart();	//restart if not work
```
Comment (snollygolly) [inline main.js:124]: Although I *do* want it to be an automatic process, I want more intelligence than just restarting.  I tried just having it restart, but wasn't pleased with the results.

File: main.js [hunk 4]
```diff
@@ -231,13 +230,15 @@ app = {
 
 	go: () => {
 		const wordLink = $(".tool-type-img").prop("src");
-		if (wordLink !== "http://s0urce.io/client/img/words/template.png") {
+		if (wordLink !== "http://www.s0urce.io/client/img/words/template.png") {
```
Comment (snollygolly) [inline main.js:233]: Why did you add the www.?

File: main.js [hunk 5]
```diff
@@ -266,7 +267,7 @@ app = {
 			url: url
 		}).done((data) => {
 			const word = String(data["ParsedResults"][0]["ParsedText"]).trim().toLowerCase().split(" ").join("");
-			if (word.length > 3) {
+			if (word.length > 2) {
```
Comment (snollygolly) [inline main.js:270]: Are there words shorter than 3?

File: db.json [hunk 6]
```diff
@@ -1,184 +1,185 @@
...
-	"http://s0urce.io/client/img/words/medium/4k2hj54.png": "export",
-  "http://s0urce.io/client/img/words/medium/32k56.png": "protocol",
-  "http://s0urce.io/client/img/words/medium/84m4.png": "fillgrid",
-  "http://s0urce.io/client/img/words/medium/5698n.png": "vector",
-  "http://s0urce.io/client/img/words/medium/jj47.png": "decryptfile",
-  "http://s0urce.io/client/img/words/medium/3j5.png": "newline",
-  "http://s0urce.io/client/img/words/medium/jkh532.png": "getid",
-  "http://s0urce.io/client/img/words/medium/jhn6n3.png": "newhost",
-  "http://s0urce.io/client/img/words/medium/k62.png": "userport"
+	"http://www.s0urce.io/client/img/words/medium/yxn24.png": "process",
```
Comment (snollygolly) [inline db.json:2]: Why did you add "www." to all these links?  In my experience they don't need it.

**Review by snollygolly — COMMENTED** [review]
It's unclear to me what changes this PR means to add, but I can't approve it in it's current form.

File: main.js (hunk 1 above)
Comment (bozoweed) [reply main.js:44]: yeah i konw i have say in my pr i have forget to change it you can make it

File: main.js (hunk 2 above)
Comment (bozoweed) [reply main.js:45]: cause /r/javascript can seen by the target and it's spoted that is a bot so for be more human i have delete that

File: main.js (hunk 3 above)
Comment (bozoweed) [reply main.js:124]: yes same for me it's for the begun next i will go debug this cause whitout this when the progress bar make a rollback due to target make +5 the bot will enter in infinity wait 
(sorry i'm french)
but try without you will seen what i want say

File: main.js (hunk 4 above)
Comment (bozoweed) [reply main.js:233]: cause i use the debug log for seen how the bot understud the link and she seen like this so make what the bot have say to me

File: main.js (hunk 5 above)
Comment (bozoweed) [reply main.js:270]: cause url is not more than 3 so the bot change target whitout finish hack

File: db.json (hunk 6 above)
Comment (bozoweed) [reply db.json:2]: cause i use the debug log for seen how the bot understud the link and she seen like this so make what the bot have say to me

Comment (bozoweed) [reply db.json:2]: i hope i have answer to all ask if you have other question tell me i will answer to it quickly

Comment (bozoweed) [reply db.json:2]: you can try all what i have say you will go to seen my update is great 
because with all my update the bot can really read and found the world ind db.json but without the bot can't found in db.json and go to search on you api and the api is little bugged
so now whit my update i'm in the top 50 in less 30 min

Comment (bozoweed) [reply db.json:2]: but please try your bot and mine you will seen mine is better for now

File: main.js (hunk 5 above)
Comment (bozoweed) [reply main.js:270]: cause com is not more than 3 
cause val is not more than 3
cause xml is not more than 3 
etc .....

**Comment by bozoweed** [comment]
but can i ask you something ? why the bot upgrade the firewall only after finish the current hack ?

File: main.js (hunk 5 above)
Comment (snollygolly) [reply main.js:270]: This is a good point.  Can you make just this change and submit it in a new PR?

Comment (bozoweed) [reply main.js:270]: why just this ? the bot is not work great with your actual code try my code and if you don't like it i will give you the right to take onli this without credit me in code

**Comment by snollygolly** [comment]
I'm only updating after the current hack because that feature is still new and I was testing it.  I intend to run it on a timer like the other loops eventually.

Also, I requested that you take your word length fix (line 270) and put it into a new PR.  I'll gladly merge it, that's an oversight on my part (thanks for the catch).  I'm not going to be merging the rest of the changes though, they seem to be matters of opinion rather than bugs or deficiencies.

**Comment by bozoweed** [comment]
it's not opinion it'sw the true your code doesn't work sorry if you don't want seen the only one true
make some test if you are a real dev make a log with world and worldlink and you will seen how you must code it bye

**Comment by snollygolly** [comment]
Your PR is kind of a mess man.  

- You changed all the DB urls to have www. in front of them, and I can't see any reason for it.  Both versions work for me.  I haven't seen any screenshots or anything indicating they don't.
- You changed the default hacking message because you didn't like it.
- You changed the DB URL to point to YOUR repository instead of mine. 

You keep saying I need to try your version because I'll see that it's better, but I don't have a problem with the current version.  It works for me, and it took me to top 5 yesterday.

I'm happy to accept pull requests that fix bugs (see the other two PRs I accepted without any debate already).  I won't be merging messy PR's that consist mostly of opinion issues and then close by calling me "not a real dev".  No thanks.

**Comment by bozoweed** [comment]
what is you web browser ?
what is your system ?
me firefox
me windows 7
and bugged exemple say text no found and go on orc to found it and return status' where is status uses where is user 
client1 where is client
and never found from db.json but whit my update i never bug again may be we need to speak about that in vocal chat on in teamviwers on my pc ( you connect on my pc )

**Comment by bozoweed** [comment]
and i have never say you are not a dev i have say "if your a real dev" so if it is you are eles you are not don't change what i say
( i dev from 6 years ago i get level master from french so don't be scare to learn with me bro )
master is more than 5 years of studie in french university
